oslo_vmware/image_transfer.py

```diff
@@ -346,3 +360,58 @@ def upload_image(context, timeout_secs, image_service, image_id, owner_id,
         updater.stop()
         read_handle.close()
     LOG.debug("Uploaded image: %s.", image_id)
+
+
+def image_pull_from_swift(swift_url, **kwargs):
+    (lease, lease_info) = \
+        rw_handles.VmdkWriteHandle._create_import_vapp_lease(
```

Comment:
I am not too happy about using private/protected functions.
At the very least, we need to make it now public to indicate the use outside of the scope. But I think, it may be worthwhile to consider the structure.

Wouldn't it make sense to add the pull functionality to `VmdkWriteHandle` perhaps, or derive a `VmdkPullHandle` from `VmdkHandle`?

Reply:
Is quite tricky here. `Handle` classes were added as different implementations over the `file` operation abstractions.
Converting the lease to Pull mode doesn't use any of the file operations.

Reply:
Let's discuss the more abstract parts in the main "thread".
On the level of those lines, if we do not change the general approach, I think the function needs to be made public.

Reply:
I would still rather see it derived. That code is not used in some code-paths is fine by me.